components/console/commands.c

```diff
@@ -209,15 +209,33 @@ esp_err_t esp_console_run(const char *cmdline, int *cmd_ret)
     return ESP_OK;
 }
 
+static struct {
+    struct arg_str *help_cmd;
+    struct arg_end *end;
+} help_args;
+
 static int help_command(int argc, char **argv)
 {
+    int nerrors = arg_parse(argc, argv, (void **) &help_args);
+
+    if (nerrors != 0) {
+        arg_print_errors(stderr, help_args.end, argv[0]);
+        return 1;
+    }
+
     cmd_item_t *it;
+    size_t len = strlen(help_args.help_cmd->sval[0]);
 
     /* Print summary of each command */
     SLIST_FOREACH(it, &s_cmd_list, next) {
         if (it->help == NULL) {
             continue;
         }
+        if (len > 0 &&
+                (strlen(it->command) != len ||
+                 strcmp(help_args.help_cmd->sval[0], it->command) != 0)) {
+            continue;
```

Comment:
Not sure there's a lot of time saved by checking strlen first and then calling strcmp... For different commands, strcmp will bail out very quickly, while strlen will always scan the command name until the end. Perhaps we can simplify things slightly by removing `size_t len` and simply using `strcmp` here?

Reply:
Removing the `size_t len`, but keeping the length check in the conditional because we still want the `help` command with no parameters to dump the entire command list